docs/source/explanations/dataclass.md

```diff
@@ -0,0 +1,455 @@
+```
+
+## Examples and Special cases
+
+### Exempting a field from events
+
+Events will *only* be emitted on value change if the name of the attribute being
+set is one of the dataclass fields (i.e. ``name in self.__annotations__``), and
+the dataclass `__post_init__` method has already been called.
+
+In the following cases, events will *not* be emitted when the value of an
+attribute is set
+1. If the field is of type
+   [`dataclasses.InitVar`](https://docs.python.org/3/library/dataclasses.html#init-only-variables)
+2. If the field is of type
+   [`typing.ClassVar`](https://docs.python.org/3/library/dataclasses.html#class-variables)
+3. If the default value of the field is a call to
+   [`dataclasses.field`](https://docs.python.org/3/library/dataclasses.html#dataclasses.field)
+   with a `metadata` dict that contains `{'events': False}`
+
+```python
+from dataclasses import InitVar, field
+from typing import ClassVar
+
+from napari.utils.events.dataclass import evented_dataclass
+
+
+@evented_dataclass
+class Q:
+    # ONLY the `a` field will emit events
+    a: int = 0
+    b: str = field(default="hi", metadata={"events": False})
+    c: InitVar[int] = 0
+    d: ClassVar[float] = 1.0
+
+
+q = Q()
+print(set(q.events._emitters))  # prints: "{'a'}"
+```
+
+Note: event emission will also be skipped if the "new" value is the same as the
+existing value
+
+### Getter/setter side-effects & conditional event logic
+
+In some cases, it may be desireable to perform additional tasks or trigger
+side-effects upon setting an attribute (in addition to emitting the basic
+event). It may also be necessary at times to provide conditional logic that
+prevents the emission of an event in specific cases.  For both of these cases,
+you may optionally define **`_on_<name>_get`** and/or **`_on_<name>_set`**
+methods that will be called when getting/setting the corresponding `<name>`
+attribute.
+
+
+```python
+from dataclasses import InitVar, field
+from typing import ClassVar
+
+from napari.utils.events.dataclass import evented_dataclass
+
+
+@evented_dataclass
+class E:
+    x: int = 1
+
+    def _on_x_get(self, value):
+        """Called when e.x is retrieved"""
+        # add optional custom logic here
+        return value
+
+    def _on_x_set(self, value):
+        """Called when `e.x = y` is set"""
+        print("setting x =", value)
+        if value > 10:
+            # return `True` to block event emission
+            return True
+        # you can override the value if you like,
+        # but use the private name to avoid recursion
+        self._x = value * 2
+
+# usage:
+# in this example, x should double the value then emit an event if val ≤ 10
+# otherwise simply set the value without emitting an event.
+
+In [1]: e = E()
+
+In [2]: e.events.x.connect(lambda e: print('x event! x =', e.value))
+
+In [3]: e.x = 3
+setting x = 3
+x event! x = 6
+
+In [4]: e.x = 15
+setting x = 15
+```
+
+### Basic type coercion with the `Property` type
```

Comment:
I might put this at the top of the special cases, or even in the main section as this is the syntax I used most commonly in our current evented dataclasses and I really like it